### PR TITLE
BaSP-TG-51: Date Picker and text Inputs hared components.

### PR DESCRIPTION
**Pull request by Licamus**
I unified date picker and text inputs, because of the continuity of the forms, and because both are inputs with labels. In all the previous codes, we take both as the same class and give them the same properties.

### CODE REVIEW

File: src/Components/Shared/TextInput/index.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import styles from './text-input.module.css';
+
+const Input = ({ children, id, name, value, onChange, type, placeholder }) => {
```
Comment (lmalvasia) [inline src/Components/Shared/TextInput/index.js:3]:
```suggestion
const TextInput = ({ label, id, name, value, onChange, type, placeholder }) => {
```

File: src/Components/Shared/TextInput/index.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  return (
+    <div className={styles.box}>
+      <label className={styles.label}>{children}</label>
```
Comment (lmalvasia) [inline src/Components/Shared/TextInput/index.js:6]:
```suggestion
      <label className={styles.label}>{label}</label>
```

File: src/Components/SuperAdmins/Form/index.js [hunk 3]
```diff
@@ -98,29 +99,48 @@ const Form = () => {
...
           <div>
-            <div className={styles.div}>
-              <label>Name</label>
-              <input type="text" name="name" value={superAdmin.name} onChange={onChange} />
-            </div>
-            <div className={styles.div}>
-              <label>Last Name</label>
-              <input type="text" name="lastName" value={superAdmin.lastName} onChange={onChange} />
-            </div>
+            <TextInput
```
Comment (lmalvasia) [inline src/Components/SuperAdmins/Form/index.js:102]:
```suggestion
            <TextInput
              label={Name}
```

File: src/Components/SuperAdmins/Form/index.js [hunk 4]
```diff
@@ -98,29 +99,48 @@ const Form = () => {
...
+              id="name"
+              name="name"
+              value={superAdmin.name}
+              onChange={onChange}
+              type="text"
+              placeholder="Name"
+            >
+              Name
```
Comment (lmalvasia) [inline src/Components/SuperAdmins/Form/index.js:110]: Remove this

File: src/Components/SuperAdmins/Form/index.js [hunk 5]
```diff
@@ -98,29 +99,48 @@ const Form = () => {
...
+            </TextInput>
```
Comment (lmalvasia) [inline src/Components/SuperAdmins/Form/index.js:111]: Remove this

File: src/Components/SuperAdmins/Form/index.js [hunk 6]
```diff
@@ -98,29 +99,48 @@ const Form = () => {
...
-              <input type="text" name="lastName" value={superAdmin.lastName} onChange={onChange} />
-            </div>
+            <TextInput
+              id="name"
+              name="name"
+              value={superAdmin.name}
+              onChange={onChange}
+              type="text"
+              placeholder="Name"
+            >
```
Comment (lmalvasia) [inline src/Components/SuperAdmins/Form/index.js:109]:
```suggestion
            />
```

File: src/Components/Shared/TextInput/index.js (hunk 1 above)
Comment (Licamus) [reply src/Components/Shared/TextInput/index.js:3]: done

File: src/Components/Shared/TextInput/index.js (hunk 2 above)
Comment (Licamus) [reply src/Components/Shared/TextInput/index.js:6]: done

File: src/Components/SuperAdmins/Form/index.js (hunk 3 above)
Comment (Licamus) [reply src/Components/SuperAdmins/Form/index.js:102]: done

File: src/Components/SuperAdmins/Form/index.js (hunk 4 above)
Comment (Licamus) [reply src/Components/SuperAdmins/Form/index.js]: done

File: src/Components/SuperAdmins/Form/index.js (hunk 5 above)
Comment (Licamus) [reply src/Components/SuperAdmins/Form/index.js]: done

File: src/Components/SuperAdmins/Form/index.js (hunk 6 above)
Comment (Licamus) [reply src/Components/SuperAdmins/Form/index.js]: done

**Review by lmalvasia — APPROVED** [review]

**Review by lmalvasia — APPROVED** [review]

**Pull request merged**